aws/aws-parallelcluster
# Implement detailed monitoring configuration

### Description of changes
* Implemented the "detailed monitoring" configuration under the "Monitoring" section.
* When enabled, the "detailed monitoring" configuration with only apply to compute nodes.
* The update policy is "COMPUTE_FLEET_STOP" for detailed monitoring.
* Change the update policy of "Monitoring" to "IGNORED" since its children layer has a mixture of update policies.
* Add validator for detailed monitoring to show a warning message that the cost might increase.

### Tests
* Manual tests
  * Three potential configurations:
    * not specified (the default is set to false)
    * explicitly set to false
    * explicitly set to true
  * Cluster creation with the above 3 configurations: worked as expected
  * Cluster update switching among these configurations: worked as expected
* Unit tests
  * Update current unit tests accordingly
  * Added unit tests for the new detailed monitoring validator

### Checklist

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
